src/main/scala/com/databricks/spark/csv/util/TypeCast.scala

```diff
@@ -44,8 +48,10 @@ object TypeCast {
         case _: ShortType => datum.toShort
         case _: IntegerType => datum.toInt
         case _: LongType => datum.toLong
-        case _: FloatType => datum.toFloat
-        case _: DoubleType => datum.toDouble
+        case _: FloatType => Try(datum.toFloat)
+          .getOrElse(java.text.NumberFormat.getInstance(locale).parse(datum).floatValue())
```

Comment:
Can't we just use ```NumberFormat.parse``` by default and avoid catching the exception? Is there a performance penalty?